testsuite/MDAnalysisTests/coordinates/test_memory.py

```diff
@@ -200,3 +200,166 @@ def test_position_assignation(self, reader):
+
+        assert not mr.ts.has_velocities
+        assert mr.ts.has_forces
+        self.assert_equal_dims(mr.ts.forces, ref_forces)
+
+    def test_both(self, ref_pos, ref_vels, ref_forces):
+        mr = MemoryReader(ref_pos,
+                          velocities=ref_vels,
+                          forces=ref_forces)
+        assert mr.ts.has_velocities
+        self.assert_equal_dims(mr.ts.velocities, ref_vels)
+        assert mr.ts.has_forces
+        self.assert_equal_dims(mr.ts.forces, ref_forces)
+
+    @pytest.mark.parametrize('param', ['velocities', 'forces'])
+    def test_wrongshape(self, ref_pos, param):
+        with pytest.raises(ValueError):
+            mr = MemoryReader(ref_pos, **{param: np.zeros((3, 2, 1))})
+
+
+class TestDimensions(object):
+    @staticmethod
+    @pytest.fixture
+    def ref_pos():
+        return np.arange(270).reshape(3, 30, 3)
+
+    @staticmethod
+    @pytest.fixture
+    def ref_box():
+        return np.arange(18).reshape(3, 6)
+
+    def test_single_box(self, ref_pos):
+        box = np.array([3, 4, 5, 90, 90, 90])
+
+        mr = MemoryReader(ref_pos, dimensions=box)
+
+        for ts in mr:
+            assert_equal(ts.dimensions, box)
+
+    def test_varying_box(self, ref_pos, ref_box):
+        mr = MemoryReader(ref_pos, dimensions=ref_box)
+
+        for i, ts in enumerate(mr):
+            assert_equal(ts.dimensions, ref_box[i])
+
+    def test_wrong_length(self, ref_pos):
+        bad_box = np.arange(12).reshape(2, 6)
+
+        with pytest.raises(ValueError):
+            mr = MemoryReader(ref_pos, dimensions=bad_box)
+
+    def test_wrong_shape(self, ref_pos):
+        bad_box = np.arange(15).reshape(3, 5)
+
+        with pytest.raises(ValueError):
+            mr = MemoryReader(ref_pos, dimensions=bad_box)
+
+
+class TestMemoryReaderModifications(object):
+    # check that modifying MR things behaves as expected
+    # in general, modifying the Timestep should be *permanent*
+    # this is unlike other Readers!
+    @staticmethod
+    @pytest.fixture()
+    def mr_reader():
+        nframes = 4
+        natoms = 10
+
+        pos = np.arange(natoms * nframes * 3).reshape(nframes, natoms, 3)
+        vel = np.arange(natoms * nframes * 3).reshape(nframes, natoms, 3) + 100
+        frc = np.arange(natoms * nframes * 3).reshape(nframes, natoms, 3) + 200
+        box = np.arange(nframes * 6).reshape(nframes, 6) + 300
+
+        return MemoryReader(pos,
+                            velocities=vel,
+                            forces=frc,
+                            dimensions=box)
+
+    @staticmethod
+    @pytest.fixture()
+    def mr_universe(mr_reader):
+        natoms = 10
+
+        u = mda.Universe.empty(natoms)
+        u.trajectory = mr_reader
+
+        return u
+
+    @pytest.mark.parametrize('attr', ['positions', 'velocities', 'forces', 'dimensions'])
+    def test_copying(self, mr_reader, attr):
+        mr2 = mr_reader.copy()
+
+        ts = mr2.ts
+        setattr(ts, attr, 7)
```

Comment:
Remind me how this works: Are you setting e.g. `ts.positions` to 7? All positions in the array or are you replacing the array with the integer "7"?

If it is the latter, shouldn't we doing the former?